src/content/docs/apm/agents/go-agent/installation/install-automation-new-relic-go.mdx

```diff
@@ -0,0 +1,99 @@
+tags:
+  - Agents
+  - Go agent
+  - Installation
+metaDescription: How to use a script to get code suggestions to instrument your Golang apps and services.
+freshnessValidatedDate: 2024-08-20 
+---
+<Callout title="preview">
+  We're still working on this feature, but we'd love for you to try it out!
+
+  This feature is currently provided as part of a preview program pursuant to our [pre-release policies](/docs/licenses/license-information/referenced-policies/new-relic-pre-release-policy).
+</Callout>
+
+Go is a compiled language with an opaque runtime, making it unable to support automatic instrumentation like other languages. For this reason, the New Relic Go agent is designed as an SDK. Since the Go agent is an SDK, it requires more manual work to set up than agents for languages that support automatic instrumentation.
+
+In an effort to reduce that manual effort, the Go agent team created the Go easy instrumentation tool, which is currently in preview. This tool will do most of the work for you by suggesting changes to your source code that instrument your application with the New Relic Go agent.
+
+To get started, check out this four-minute video, or skip down to [How it works](#how-it-works):
+
+[![asciicast](https://asciinema.org/a/r0Il7o2eMiZaLKHIlew3IL2nx.svg)](https://asciinema.org/a/r0Il7o2eMiZaLKHIlew3IL2nx)
+
+## Preview notice [#preview-notice]
+
+This feature is currently provided as part of a product preview and is subject to our New Relic Experimental policies. Recommended code changes are suggestions only and should be subject to human review for accuracy, applicability, and appropriateness for your environment. This feature should only be used in non-critical, non-production environments that do not contain sensitive data.
+
+This project, its code, and the UX are under heavy development, so you should expect it to change. Please take this into consideration when participating in this preview. If you encounter any issues, please report them using [Github issues](https://github.com/newrelic/go-agent-assisted-instrumentation) and fill out as much of the issue template as you can so we can improve this tool.
+
+## How it works [#how-it-works]
+
+This tool will not interfere with your application's operation, and it doesn't make any changes to your code directly. Here's what happens:
+
+* It analyzes your source code and identifies opportunities to instrument it.
+* It suggests changes to your code that use the New Relic Go agent SDK to capture telemetry data.
+* You review those sugestions that are inserted in a `.diff` file and decide which changes to apply to your source code.
+
+As part of the analysis, this tool may invoke `go get` or other Go language toolchain commands which may modify your `go.mod` file, but not your actual source code.
+
+<Callout variant="important">
+  This tool can't detect if you already have New Relic instrumentation. Please only use this on applications without any instrumentation.
```

Comment:
Use the tool on applications without any instrumentation.

Reply:
Nice suggestion! I'll fix this!